docs/metrics/places/metrics.md

```diff
@@ -0,0 +1,31 @@
+<!-- AUTOGENERATED BY glean_parser.  DO NOT EDIT. -->
```

Comment:
Does this file need to be checked in? In a perfect world, we'd probably have our CI generate it and stash it somewhere useful.

Reply:
[The ticket](https://github.com/mozilla/application-services/issues/2300) says so 😬. Re: adding a glean parser build step, I'm torn. I made a few changes to the `metrics.yaml` file while creating this PR and waiting for the build to finish to ensure that the docs are accurate seems like a big feedback loop. But if you're certain of the metrics you want to create maybe not.

Reply:
It's probably fine to go with, but if it doesn't land before Ryan is back I'll ask him. Note also that I've no issue with ensuring that file is generated locally, I'm just not sure it needs to be checked in. For example, if someone makes a trivial typo fix to that file, there seems a chance they will check the source in without the generated file, which will be confusing for others. But it's not a big deal and certainly not a blocker.

Reply:
Definitely a worthwhile conversation to have. In general I think we need to discuss conventions for adding glean metrics to components so that we're efficiently creating and consolidating code and docs.

Reply:
I think we should check it in for now in order to ensure it's easy to find, but hopefully as the glean pipeline tooling continues to develop there will be easier ways to manage this.

Reply:
Hey, just saw this conversion and I thought I'd drop by :-) Within Mozilla, all the chunks of code performing collection are *required* to have documentation about the data they are collecting. That is true for both [legacy telemetry](https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/data/index.html) and the Glean SDK.

Prior to Glean, developers were required to manually create a human readable document (wiki, rst or else). The Glean SDK makes it a bit easier, by automatically generating a Markdown document for you, in-tree. By being in-tree this is guaranteed to be in sync with whatever your component is advertising and collecting through the Glean registry files (`metrics.yaml` and `pings.yaml`). It also has the nice advantage of being versioned the same way as your registry files, allowing to walk back in history if required and easing "code-archeology".

> For example, if someone makes a trivial typo fix to that file, there seems a chance they will check the source in without the generated file, which will be confusing for others.

This happened before and it tends to auto-fix very easily: whoever builds the tree the next time will see a change for the documentation file and commit that too.

Having CI solutions requires CI-tinkering which is not a long-term, affordable solution for all the Mozilla teams. Moreover, you loose the advantage of versioning documentation.

With that said, this is a built-time tool we offer, but is not mandatory :)